src/Discovery/DiscoveryDetails/DataDownloadList/ActionButtons/DownloadUtils/DownloadAllFiles.tsx

```diff
@@ -0,0 +1,165 @@
+) => {
+  fetchWithCreds({ path: `${jobAPIPath}status?UID=${uid}` }).then(
+    (statusResponse) => {
+      const { status } = statusResponse.data;
+      if (statusResponse.status !== 200 || !status) {
+        // usually empty status message means Sower can't find a job by its UID
+        setDownloadStatus(DOWNLOAD_FAIL_STATUS);
+      } else if (status === 'Failed') {
+        fetchWithCreds({ path: `${jobAPIPath}output?UID=${uid}` })
+          .then((outputResponse) => {
+            const { output } = outputResponse.data;
+            if (outputResponse.status !== 200 || !output) {
+              setDownloadStatus(DOWNLOAD_FAIL_STATUS);
+            } else {
+              setDownloadStatus({
+                inProgress: false,
+                message: {
+                  title: 'Download failed',
+                  content: <p>{output}</p>,
+                  active: true,
+                },
+              });
+            }
+          })
+          .catch(() => setDownloadStatus(DOWNLOAD_FAIL_STATUS));
+      } else if (status === 'Completed') {
+        fetchWithCreds({ path: `${jobAPIPath}output?UID=${uid}` })
+          .then((outputResponse) => {
+            const { output } = outputResponse.data;
+            if (outputResponse.status !== 200 || !output) {
+              setDownloadStatus(DOWNLOAD_FAIL_STATUS);
+            } else {
+              try {
+                const regexp = /^https?:\/\/(\S+)\.s3\.amazonaws\.com\/(\S+)/gm;
+                if (!new RegExp(regexp).test(output)) {
+                  throw new Error('Invalid download URL');
+                }
+                setDownloadStatus({
+                  inProgress: false,
+                  message: {
+                    title: 'Your download is ready',
+                    content: (
+                      <React.Fragment>
+                        <p> {DOWNLOAD_SUCCEEDED_MESSAGE} </p>
+                        <a href={output} target='_blank' rel='noreferrer'>
+                          {output}
+                        </a>
+                      </React.Fragment>
+                    ),
+                    active: true,
+                  },
+                });
+                setTimeout(() => window.open(output), 2000);
+              } catch {
+                // job output is not a url -> is an error message
+                setDownloadStatus({
+                  inProgress: false,
+                  message: {
+                    title: 'Download failed',
+                    content: <p>{output}</p>,
+                    active: true,
+                  },
+                });
+              }
+            }
+          })
+          .catch(() => setDownloadStatus(DOWNLOAD_FAIL_STATUS));
+      } else {
+        setTimeout(
+          checkDownloadStatus,
+          JOB_POLLING_INTERVAL,
+          uid,
+          downloadStatus,
+          setDownloadStatus,
+        );
+      }
+    },
+  );
+};
+
+const DownloadAllFiles = (
```

Comment:
We probably should add above-mentioned `checkFederatedLoginStatus()` and `healRequiredIDPLogic()` related logics from https://github.com/uc-cdis/data-portal/blob/master/src/Discovery/DiscoveryActionBar.tsx into here

Reply:
Note: todo Dec. 7th -> address missing checkFederatedLoginStatus() here.